.github/workflows/stack-integration_tests.yml

```diff
@@ -118,7 +118,102 @@ jobs:
       #   shell: bash
       #   run: |
       #     ./scripts/docker_save.sh
+  
+  stack-integration-tests-windows:
+    strategy:
+      max-parallel: 3
+      matrix:
+        os: [Windows-10]
+        python-version: [3.9]
+
+    runs-on: [self-hosted,windows]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check for file changes
+        uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        if: steps.changes.outputs.stack == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Get pip cache dir
+        if: steps.changes.outputs.stack == 'true'
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: pip cache
+        uses: actions/cache@v2
+        if: steps.changes.outputs.stack == 'true'
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-py${{ matrix.python-version }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ matrix.python-version }}
+
+      - name: Upgrade pip
+        if: steps.changes.outputs.stack == 'true'
+        run: |
+          pip install --upgrade --user pip
 
+      - name: Install tox
+        if: steps.changes.outputs.stack == 'true'
+        run: |
+          pip install tox --upgrade
+
+      - name: Install Docker Compose
+        if: runner.os == 'Windows'
+        run: |
+          curl -L https://github.com/docker/compose/releases/download/v2.1.1/docker-compose-windows-x86_64.exe -o "C:\Program Files\Docker\cli-plugins\docker-compose.exe"
+
+      - name: Setup WSL
```

Comment:
this wont work, we need to pre-provision WSL2 and desktop docker on these machines. Once its working we can always create an image so we can destroy and re-create the machine without the setup work. This is also super handy for quickly getting it setup:
https://raw.githubusercontent.com/madhavajay/PySyft/madhava/win_bootstrap/packages/hagrid/hagrid/win_bootstrap.py

Reply:
Thanks @madhavajay ! I wasn't going to depend on it actually, I just wanted to split out the windows CI test so it could fail separately.